pkg/evaluators/identity/api_key.go

```diff
@@ -134,7 +144,6 @@ func (a *APIKey) RevokeK8sSecretBasedIdentity(ctx context.Context, deleted k8s_t
 		if secret.GetNamespace() == deleted.Namespace && secret.GetName() == deleted.Name {
 			delete(a.secrets, key)
 			log.FromContext(ctx).WithName("apikey").V(1).Info("api key deleted")
-			return
```

Comment:
Should no longer return early since the change now allows for the possibility of multiple keys pointing to the same secret